API/Routes/authenticationRoute.ts

```diff
@@ -0,0 +1,172 @@
+import {Request, Response, Router} from "express";
+import bcrypt from 'bcryptjs';
+import crypto from "crypto";
+import {authenticate} from "../authenticationMiddleware.js";
+import {User} from "../../Database/Mapper/Entities/user.js";
+import {Session} from "../../Database/Mapper/Entities/session.js";
+import {Permission_Techcode} from "../../Database/Mapper/Techcodes/Permission_Techcode";
```

Comment:
.js

Reply:
that would have been an anoying bug to find...